kayak_core/src/styles/mod.rs

```diff
@@ -141,23 +155,91 @@ macro_rules! define_styles {
 }
 
 define_styles! {
+
+    /// A struct used to define the look of a widget
+    ///
+    /// # Example
+    ///
+    /// All fields are `pub`, so you can simply define your styles like:
+    ///
+    /// ```
+    /// # use kayak_core::styles::{Style, StyleProp, Units};
+    /// let style = Style {
+    ///   width: StyleProp::Value(Units::Pixels(100.0)),
+    ///   // Or using `into()` to convert to `StyleProp`
+    ///   // width: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    /// ```
+    ///
+    /// You can also create styles using other styles in a builder-like syntax:
+    ///
+    /// ```
+    /// # use kayak_core::styles::{Style, StyleProp, Units};
+    /// let style_a = Style {
+    ///   width: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    /// let style_b = Style {
+    ///   height: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    ///
+    /// let style = Style::default() // <- Initializes all fields as `StyleProp::Unset`
+    ///   // Applied first (sets any `StyleProp::Unset` fields)
+    ///   .with_style(&style_a)
+    ///   // Applied second (sets any remaining `StyleProp::Unset` fields)
+    ///   .with_style(&style_b);
+    /// ```
     #[derive(Debug, Default, Clone, PartialEq)]
     pub struct Style {
```

Comment:
Should we also add a [`#[non_exhaustive]`](https://doc.rust-lang.org/reference/attributes/type_system.html) attribute? I imagine this struct could change a fair amount more (although I hope no one is manually writing out all fields...).

Reply:
Lets do that. :)

Reply:
Nevermind, just tried it and apparently it doesn't support the update syntax. PReventing us from doing:

```rust
let styles = Style {
  color: Color::BLACK.into(),
  ..Default::default()
}
```

And instead forcing us to do:

```rust
let mut styles = Style::default();
styles.color = Color::BLACK.into();
```

Which makes the new builder patterns a little more awkward to use:

```rust
let mut user_styles = Style::default();
user_styles.color = Color::BLACK.into();
// ...
```

Reply:
Just keep it as is! 👍